app/grandchallenge/components/utils.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+from grandchallenge.components.models import InterfaceKindChoices
+from grandchallenge.core.validators import JSONValidator
+from grandchallenge.hanging_protocols.models import (
+    VIEW_CONTENT_SCHEMA,
+    ViewportNames,
+)
+
+
+def generate_view_content_example(interfaces):
+    images = list(
+        interfaces.filter(kind=InterfaceKindChoices.IMAGE)
+        .order_by("slug")
+        .values_list("slug", flat=True)
+    )
+
+    if not images:
+        return None
+
+    overlays = list(
+        interfaces.filter(
+            kind__in=[
+                InterfaceKindChoices.SEGMENTATION,
+                InterfaceKindChoices.HEAT_MAP,
+                InterfaceKindChoices.DISPLACEMENT_FIELD,
+            ]
+        )
+        .order_by("slug")
+        .values_list("slug", flat=True)
+    )
+    view_content_example = {}
+    overlays_per_image = len(overlays) // len(images)
+    remaining_overlays = len(overlays) % len(images)
+
+    for port in ViewportNames.values:
+        if len(images) == 0:
+            break
```

Comment:
You are already checking if images is empty above, so no need to repeat that here, right?

Reply:
This is for the for loop, I am looping over the viewport names which can be more than the number of available images and poping the image from the list. And we need to finish when image_ci list is empty, that is what this check does.

Reply:
Ah of course, thanks for clarifying!